nucleus/annotation.py

```diff
@@ -15,9 +15,86 @@
     LABEL_KEY,
     TYPE_KEY,
     VERTICES_KEY,
+    ITEM_ID_KEY,
+    MASK_URL_KEY,
+    INDEX_KEY,
+    ANNOTATIONS_KEY,
 )
 
 
+class Segment:
+    def __init__(
+        self, label: str, index: int, metadata: Optional[dict] = None
+    ):
+        self.label = label
+        self.index = index
+        self.metadata = metadata
+
+    def __str__(self):
+        return str(self.to_payload())
+
+    @classmethod
+    def from_json(cls, payload: dict):
+        return cls(
+            label=payload.get(LABEL_KEY, ""),
+            index=payload.get(INDEX_KEY, None),
+            metadata=payload.get(METADATA_KEY, None),
+        )
+
+    def to_payload(self) -> dict:
+        payload = {
+            LABEL_KEY: self.label,
+            INDEX_KEY: self.index,
+        }
+        if self.metadata is not None:
+            payload[METADATA_KEY] = self.metadata
+        return payload
+
+
+class SegmentationAnnotation:
+    def __init__(
+        self,
+        mask_url: str,
+        annotations: List[Segment],
+        reference_id: Optional[str] = None,
+        item_id: Optional[str] = None,
```

Comment:
Looks like this API is missing support for an `annotation_id`, which we should be consistent about across annotation types

Reply:
(I know there isn't an idempotency benefit in this case because there can only be one semseg mask per image, unlike bounding boxes, but I think it still makes sense to get users in the habit of always providing annotation_ids with their annotations and have consistency across our API. We will want to support querying by annotation_id directly in the query bar down the line, e.g., which would be relevant for semseg masks too. Also useful for when we eventually support GT versioning.)

Reply:
I would recommend saving the annotation_id for a separate follow-up PR because the changes required are mostly on the scaleapi side.  I imagine since uniqueness is already enforced on dataset items, the main value prop is annotation_ids would not allow you upload same mask to two different items

Reply:
Sgtm